src/components/etherpad.jsx

```diff
@@ -52,18 +61,27 @@ const EtherpadInner = createReactClass({
     // from https://github.com/ether/etherpad-lite-jquery-plugin/blob/master/js/etherpad.js
     return `${hostName}/p/${padName}`;
   },
+  getEtherpadInfo() {
+    const reg = /^(?:([^:\/?#]+):)?(?:\/\/([^\/?#]*):([0-9]+))?([^?#]*)(?:\?([^#]*))?(?:#(.*))?/;
+    let m = window.localStorage.getItem('ep-url').match(reg);
```

Comment:
Agree, but that can happen when we do other bug fixes or enhancements to this component.

There will be lots more things to do here, especially deactivating etherpad in other components, especially when etherpad settings are missing, or the token fails, etc

Reply:
Yes, but we will need to prevent this component loading if it is not set correctly.  It is useless fixing this here.

`EtherpadClient.connect(self.getEtherpadInfo())` is going to fail if this doesnt return a valid connection metadata.

e.g. if I put in `foo` as the URL setting.

Ideally the validation of the user setting is moved to the modal.
But this PR gives credit where it is due, and lets us build on top.

First step is disabling the editor everywhere if the setting is empty.

Then the problems of this entire component are much less critical as the user cant encounter the problems in the first few clicks.